BinaryTree.java

```diff
@@ -0,0 +1,165 @@
+package com.katermar;
+
+/**
+ * Created by katermar on 5/23/2017.
+ */
+public class BinaryTree {
+    TreeNode root;
+
+    /*Inner class for Binary Tree node
+    * @field value stores integer value
+    * @field left, right store left and right subtrees*/
+    public static class TreeNode {
+        private int value;
+        private TreeNode left, right;
+
+        TreeNode(int item) {
+            value = item;
+            left = right = null;
+        }
+
+        public int getValue() {
+            return value;
+        }
+
+        public TreeNode getLeft() {
+            return left;
+        }
+
+        @Override
+        public String toString() {
+            return "TreeNode{" +
+                    "value=" + value +
+                    '}';
+        }
+
+        public TreeNode getRight() {
+            return right;
+        }
+
+    }
+
+    /* Function to print all ancestors of purposed value
+    * @param node - root node
+    * @param value - value of the node on which we want to print ancestors
+    * @return true, if value exists in this tree, otherwise - false*/
+    boolean printAllAncestors(TreeNode node, int value)

```

Comment:
What if we at some point we need to do something useful (rather than just print) with node's ancestors? Or test it more extensively in automated fashion?

Could you please refactor code so that finding them is separate from printing them?